File: Exemplos/App_eXperience_Flutter/Elgin_AppExperience_Flutter/android/app/src/main/java/com/elgin/flutter_m8/TefEnums/Acao.java
package com.elgin.flutter_m8.TefEnums;

//Ações disponíveis, correspondente aos botões na tela.
public enum Acao {
    VENDA("venda"), CANCELAMENTO("cancelamento"), CONFIGURACAO("configuracao");

    //String rotulo utilizado para aferir os parâmetros enviados a partir da plataforma Flutter.
    private final String rotulo;

    Acao(String rotulo) {
        this.rotulo = rotulo;
    }

    public static Acao fromRotulo(String rotulo) {
        for (Acao acao : values()) {
            if (acao.rotulo.equals(rotulo)) {
                return acao;
            }
        }
        throw new IllegalArgumentException("Nenhuma ação válida enviada!");
    }
}

File: Exemplos/App_eXperience_Flutter/Elgin_AppExperience_Flutter-font/lib/pages/tef/Acao.dart
part of './TefPage.dart';

//Ações disponíveis nos TEFs.
class Acao {
  static const Acao VENDA  = Acao._('venda');
  static const Acao CANCELAMENTO = Acao._('cancelamento');
  static const Acao CONFIGURACAO = Acao._('configuracao');

  //String rotulo utilizada para a decomposição do enum e captura do parâmetro na plataforma Android.
  final String _rotulo;

  const Acao._(this._rotulo);

  @override
  toString() => _rotulo;
}